pkg/sdk/testint/secrets_gen_integration_test.go

```diff
@@ -0,0 +1,716 @@
+package testint
+
+import (
+	"testing"
+	"time"
+
+	assertions "github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance/bettertestspoc/assert"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance/bettertestspoc/assert/objectassert"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/acceptance/helpers/random"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestInt_Secrets(t *testing.T) {
+	client := testClient(t)
+	ctx := testContext(t)
+
+	integrationId := testClientHelper().Ids.RandomAccountObjectIdentifier()
+
+	// "YYYY-MM-DD" or "YYYY-MM-DD HH-MI-SS" format has to be used, otherwise Snowflake returns error: "Invalid date/time format"
+	refreshTokenExpiryTime := time.Now().Add(24 * time.Hour).Format(time.DateOnly)
+
+	_, apiIntegrationCleanup := testClientHelper().SecurityIntegration.CreateApiAuthenticationClientCredentialsWithRequest(t,
+		sdk.NewCreateApiAuthenticationWithClientCredentialsFlowSecurityIntegrationRequest(integrationId, true, "foo", "foo").
+			WithOauthAllowedScopes([]sdk.AllowedScope{{Scope: "foo"}, {Scope: "bar"}}),
+	)
+	t.Cleanup(apiIntegrationCleanup)
+
+	stringDateToSnowflakeTimeFormat := func(inputLayout, date string) *time.Time {
```

Comment:
I'm not sure, but i think that it would fail. If there is a way to change location of the account then it would be nice to test it and ultimately change the test so it won't depend on account location somehow.